src/pools/D3MAaveDaiPool.sol

```diff
@@ -0,0 +1,201 @@
+        asset = TokenLike(dai_);
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, ,) = LendingPoolLike(pool_).getReserveData(dai_);
+        require(adai_ != address(0), "D3MAaveDaiPool/invalid-adai");
+        require(stableDebt_ != address(0), "D3MAaveDaiPool/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDaiPool/invalid-variableDebt");
+
+        adai = ATokenLike(adai_);
+        stableDebt = ATokenLike(stableDebt_);
+        variableDebt = ATokenLike(variableDebt_);
+        rewardsClaimer = RewardsClaimerLike(_rewardsClaimer);
+
+        ATokenLike(adai_).approve(pool_, type(uint256).max);
+        TokenLike(dai_).approve(pool_, type(uint256).max);
+
+        CanLike(d3mHubLike(hub_).vat()).hope(hub_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Math ---
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, "D3MAaveDaiPool/overflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPool/overflow");
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, address data) external auth {
+        if (what == "king") king = data;
+        else revert("D3MAaveDaiPool/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // Deposits Dai to Aave in exchange for adai which gets sent to the msg.sender
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit
+    function deposit(uint256 amt) external override auth {
+        uint256 scaledPrev = adai.scaledBalanceOf(address(this));
+
+        pool.deposit(address(asset), amt, address(this), 0);
+
+        // Verify the correct amount of adai shows up
+        uint256 interestIndex = pool.getReserveNormalizedIncome(address(asset));
+        uint256 scaledAmount = _rdiv(amt, interestIndex);
+        require(adai.scaledBalanceOf(address(this)) >= _add(scaledPrev, scaledAmount), "D3MAaveDaiPool/incorrect-share-credit");
+    }
+
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
+    function withdraw(uint256 amt) external override auth {
+        pool.withdraw(address(asset), amt, address(msg.sender));
+    }
+
+    // --- Collect any rewards ---
+    function collect() external returns (uint256 amt) {
+        require(king != address(0), "D3MAaveDaiPool/king-not-set");
+
+        address[] memory assets = new address[](1);
+        assets[0] = address(adai);
+
+        uint256 amount = type(uint256).max;
```

Comment:
Can inline this, wull also avoid the usage of both `amount` and `amt`.

Reply:
I updates the `amount` to be line, but I'm not sure I see how we remove `amt`. My understanding is that that is how much we actually collected. let me know if i missed something

Reply:
I meant the fact that we have both a variable called `amt` and one that's called `amount`.
Anyway that is no longer the case and the code is now ok.